entities/modulecapabilities/offload.go

```diff
@@ -0,0 +1,35 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modulecapabilities
+
+import (
+	"context"
+)
+
+type OffloadCloud interface {
+	// Upload uploads the content of a shard to s3
+	// upload shard content assigned to specific node
+	// s3://{configured_bucket}/{className}/{shardName}/{nodeName}/{shard content}
+	Upload(ctx context.Context, className, shardName, nodeName string) error
+	// Download uploads the content of a shard to s3
+	// download s3 bucket content to desired node
+	// {dataPath}/{className}/{shardName}/{content}
+	Download(ctx context.Context, className, shardName, nodeName string) error
+	// Delete deletes content of a shard to s3
+	// upload shard content assigned to specific node
+	// s3://{configured_bucket}/{className}/{shardName}/{nodeName}/{shard content}
+	Delete(ctx context.Context, className, shardName, nodeName string) error
+}
+
+type OffloadProvider interface {
+	OffloadBackend(backend string) (OffloadCloud, error)
+}
```

Comment:
I think that this interface can be removed or declared only in migrator package. In modulecapabilities we only put the interfaces that describe modules, we don't put in any place the interfaces that describe the provider capabilities, we only put those Provider like interfaces in the code directly where we inject module provider